core/src/test/scala/org/apache/spark/BarrierStageOnSubmittedSuite.scala

```diff
@@ -185,4 +185,56 @@ class BarrierStageOnSubmittedSuite extends SparkFunSuite with LocalSparkContext
     testSubmitJob(sc, rdd,
       message = DAGScheduler.ERROR_MESSAGE_RUN_BARRIER_WITH_DYN_ALLOCATION)
   }
+
```

Comment:
We need a test verifies if total slots are good but some are running other jobs, we shouldn't fail the barrier job.